### PR TITLE
LPS-140019 Performance improvements for Subtype and Mime Type Treefilter

### PR DESCRIPTION
**Pull request by beltranrengifo**
[Jira issue](https://issues.liferay.com/browse/LPS-140019)

### Motivation & Context
[Jira spike ticket](https://issues.liferay.com/browse/LPS-138166)
[Confluence documentation](https://liferay.atlassian.net/wiki/spaces/ENGTANGO/pages/1890844921/LPS-138166+-+Frontend+performance+tests+for+the+tree+filter+Subtype+and+Mime+Type)

### How to test
- Navigate to the Content Dashboard
- Filter by Type or Extension
- Perform a search
- Search process should now be debounced by 500ms
- Parent nodes after the search should be NOT expanded, except the first one
- The React Profiler should show better results, in timings and renders, than executing the same processes in master branch

### CODE REVIEW

File: modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/TreeFilter.js [hunk 1]
```diff
@@ -26,6 +27,9 @@ import {
 	visit,
 } from '../utils/tree-utils';
 
+const SEARCH_QUERY_MIN_LENGHT = 2;
+const SEARCH_INPUT_DEBOUNCE = 500;
```
Comment (claraizquierdo) [inline modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/TreeFilter.js:31]: This value is a little bit high and makes the filtering look slow. I think that 200 would be a better value, don't you think?

Comment (beltranrengifo) [reply modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/TreeFilter.js:31]: Totally agree. Thxs

**Comment by beltranrengifo** [comment]
ci:test:sf

**Review by claraizquierdo — APPROVED** [review]

**Comment by beltranrengifo** [comment]
ci:test:relevant

**Comment by beltranrengifo** [comment]
ci:forward

**Pull request closed without merging**